<!-- move base -->
<launch>
  <master auto="start"/>
    <arg name="visualize" default="false"/>

<!-- <param name="use_sim_time" value="true" /> -->

  <node pkg="move_base" type="trajectory_recode_node" respawn="false" name="trajectory_recode_node" output="screen"/>

  <node pkg="move_base" type="local_tracker_node" respawn="false" name="local_tracker_node" output="screen">
    <param name="controller_frequency" value="5.0" />
    <param name="controller_patience" value="1.0" /> 
    <!-- <param name="oscillation_timeout" value="30.0" /> -->
    <!-- <param name="oscillation_distance" value="0.5" /> -->
      <param name="oscillation_timeout" value="5.0" />
    <param name="oscillation_distance" value="0.1" />
    <remap from="cmd_vel" to="mobile_base/commands/velocity"/>

    <rosparam file="$(find move_base)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find move_base)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find move_base)/param/base_local_planner_params.yaml" command="load" />  

    <param name="global_costmap/global_frame" value="map"/>
    <param name="global_costmap/robot_base_frame" value="base_footprint"/>
    <param name="local_costmap/global_frame" value="odom"/>
    <param name="local_costmap/robot_base_frame" value="base_footprint"/>
    <param name="local_costmap/laser_scan_sensor/sensor_frame" value="base_laser_link"/>
    <param name="local_costmap/laser_scan_sensor/topic" value="/scan"/>
    <param name="local_costmap/obstacle_layer/laser_scan_sensor/topic" value="/scan"/>
  </node>

   <node if="$(arg visualize)" name="rvizvisualisation" pkg="rviz" type="rviz" output="log" args="-d $(find move_base)/rviz/turtlebot_tracker.rviz" />
</launch>
